src/components/Popups/ReplyField/ReplyField.tsx

```diff
@@ -64,19 +122,78 @@ export default class ReplyField extends React.Component<Props, State> {
         setCursorPosition(editorState.getSelection().getFocusOffset());
     };
 
+    handleMention = (): void => {
+        const { onMention } = this.props;
+        const { activeMention } = this.state;
+
+        if (onMention && activeMention) {
+            onMention(activeMention.mentionString);
+        }
+    };
+
+    getVirtualElement = (activeMention: Mention): VirtualElement | null => {
+        const selection = window.getSelection();
+        if (!selection?.focusNode) {
+            return null;
+        }
+
+        const range = selection?.getRangeAt(0);
+        const textNode = range.endContainer;
+
+        range.setStart(textNode, activeMention.start);
+        range.setEnd(textNode, activeMention.start);
+
+        const mentionRect = range.getClientRects()[0];
+
+        const virtualElement = {
+            getBoundingClientRect: () => mentionRect,
+        };
+
+        // restore selection
+        range.setStart(textNode, activeMention.end);
+        range.setEnd(textNode, activeMention.end);
+
+        return virtualElement;
+    };
+
+    updatePopup = (): void => {
+        const { activeMention } = this.state;
+
+        if (!activeMention) {
+            this.hidePopup();
+            return;
+        }
+
+        this.setState({
+            popupReference: this.getVirtualElement(activeMention),
```

Comment:
It seems like we could derive this information entirely in render rather than storing it in state.